src/services/identity/UsersService.ts

```diff
@@ -188,26 +210,106 @@ class UsersService {
   }
 
   async sendEmailOtp(email: string) {
-    const otp = this.otp.generate(email)
-    const expiry = this.otp.getExpiryMinutes()
+    const { otp, hashedOtp } = await this.otpService.generateLoginOtpWithHash()
+
+    // Reset attempts to login
+    const otpEntry = await this.otpRepository.findOne({ where: { email } })
+    if (!otpEntry) {
+      // create new entry
+      await this.createOtpEntry(email, OtpType.Email, hashedOtp)
+    } else {
+      await otpEntry?.update({
+        hashedOtp,
+        attempts: 0,
+        expiresAt: this.getOtpExpiry(),
+      })
+    }
```

Comment:
feels like this should belong inside the `OtpService` for better separation of concerns; wdyt?

Reply:
Hmm, it was intentional to keep OtpService agnostic of the biz logic involved here - e.g. to find by email and then performing a reset. This operation can be summarised as "reset the attempts for an user" which is under the UsersService's purview

The OtpService was modelled to provide abstractions over the Otp operations - generating and verifying Otps. If you realise OtpService does not interact with OtpRepository. It merely provides operations required for the creations/verifications.

Reply:
i understand where you're coming from but seeing `UsersService` call `otpService.generateLoginOtpWithHash` and then `otpRepository.findOne` feels kinda weird @__@

i'm more inclined towards the NestJS way of gating the repository behind its service but wanted to hear your thouhghts on this

Reply:
I think we need a revamp of the structure tbh.

- routers -> handlers -> services -> repositories -> models

That should be the flow.

Also, regarding: 

> seeing UsersService call otpService.generateLoginOtpWithHash and then otpRepository.findOne feels kinda weird

We should see services and repositories as like lego blocks that other services and repos can tap on. Meaning to say, OtpRepository is not meant to be only consumed by OtpService.

The service layer which is performing the biz logic should use the underlying blocks to execute the biz logic. UserService can interact with other services or utils.

But if you feel it's weird still, we can do away with OtpService in favour of OtpUtils or something haha. So that we don't see it as a service layer

wdyt?